chain-db-watcher/src/sync/index.ts

```diff
@@ -64,15 +64,24 @@ const syncTreasuryProposals = async (onchainTreasuryProposals: ObjectMap, discus
 	}));
 };
 
-const syncMotions = async (onchainMotions: MotionObjectMap, discussionMotions: ObjectMap): Promise<void[]> => {
+const syncMotions = async (onchainMotions: MotionObjectMap, discussionMotions: ObjectMap, treasuryDeduplicatedMotionsMap: TreasuryDeduplicateMotionMap): Promise<void[]> => {
 	return Promise.all(Object.entries(onchainMotions).map(async ([key, val]: [string, OnchainMotionSyncType]) => {
+		// the same treasury id might have several motions attached to it.
+		// only the last motion should be added to the discussion db
+
 		// if this motion doesn't exist in the discussion DB
 		if (!discussionMotions[key]) {
-			if (val.treasuryProposalId || val.treasuryProposalId === 0) {
-				await updateTreasuryProposalWithMotion({
-					onchainMotionProposalId: Number(key),
-					onchainTreasuryProposalId: val.treasuryProposalId
-				});
+			const motionFollowsTreasury = val.treasuryProposalId || val.treasuryProposalId === 0;
+			// if this motion is associated to a treasury proposal
+			if (motionFollowsTreasury) {
+				const latestMotion = Math.max(...treasuryDeduplicatedMotionsMap[val.treasuryProposalId as number]);
```

Comment:
this "as number" is not a hack, `val.treasuryProposalId` is a number, because it's inside a `if (motionFollowsTreasury)` that could be written as `if (val.treasuryProposalId || val.treasuryProposalId === 0)`... but the TS compiler doesn't get it :(

Reply:
Changed it to make it less readable.. but with comments it should be ok and removing that ugly `as`